src/routes/exception.ts

```diff
@@ -19,19 +19,29 @@
 
 import * as express from 'express';
 import multer from 'multer';
-import exceptionApi from '../exception/exception-api';
+import exceptionApi, { parseTSV, requestContainsFile } from '../exception/exception-api';
+import { isEntityExceptionRecord, isProgramExceptionRecord } from '../exception/types';
 import { wrapAsync } from '../middleware';
 
+// config
 const router = express.Router({ mergeParams: true });
-
 const upload = multer({ dest: '/tmp' });
 
+// routes
+router.post('*', upload.single('exceptionFile'), requestContainsFile);
+
 router.post(
   '/',
-  upload.single('programExceptionFile'),
+  parseTSV(isProgramExceptionRecord),
```

Comment:
Using parseTSV here as middleware instead of in the service I think causes us to lose our type safety... see other comments